### PR TITLE
Removed monitoring role as no longer needed

### PR DESCRIPTION
**Pull request by robshelly**
### Type of change

- Bugfix

### Description

Removed the monitoring ansible role as the monitoring resources are now installed by the operator

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cmd/controller-manager/main.go [hunk 1]
```diff
@@ -211,7 +211,9 @@ func main() {
 	}
 
 	// Add the Metrics Service
-	addMetrics(ctx, cfg, namespace)
+	if monitoringEnabled == "true" {
```
Comment (k-wall) [inline cmd/controller-manager/main.go:214]: Suggest we change above to use:

	monitoringEnabled := util.GetBooleanEnv("ENABLE_MONITORING")

File: pkg/controller/consoleservice/consoleservice_controller.go [hunk 2]
```diff
@@ -463,39 +463,44 @@ func applyService(consoleService *v1beta1.ConsoleService, service *corev1.Servic
 
 func (r *ReconcileConsoleService) reconcilePrometheusRule(ctx context.Context, consoleservice *v1beta1.ConsoleService) (reconcile.Result, error) {
 
-	if !util.HasApi(monitoringGVK) {
-		return reconcile.Result{}, nil
-	}
+	monitoringEnabled := os.Getenv("ENABLE_MONITORING")
```
Comment (k-wall) [inline pkg/controller/consoleservice/consoleservice_controller.go:466]:
```suggestion
	monitoringEnabled := util.GetBooleanEnv("ENABLE_MONITORING")
```

File: pkg/controller/consoleservice/consoleservice_controller.go [hunk 3]
```diff
@@ -463,39 +463,44 @@ func applyService(consoleService *v1beta1.ConsoleService, service *corev1.Servic
...
 
-	prometheusRule := &monitoringv1.PrometheusRule{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "enmasse-console-rules",
-			Namespace: consoleservice.Namespace,
-		},
-	}
+	if monitoringEnabled == "true" {
```
Comment (k-wall) [inline pkg/controller/consoleservice/consoleservice_controller.go:468]:
```suggestion
	if monitoringEnabled  {
```

File: pkg/controller/consoleservice/consoleservice_controller.go [hunk 4]
```diff
@@ -463,39 +463,44 @@ func applyService(consoleService *v1beta1.ConsoleService, service *corev1.Servic
...
-		install.ApplyCustomLabel(&prometheusRule.ObjectMeta, "monitoring-key", "middleware")
-		prometheusRule.Spec = monitoringv1.PrometheusRuleSpec{
-			Groups: []monitoringv1.RuleGroup{
-				{
-					Name: "ComponentHealth",
-					Rules: []monitoringv1.Rule{
-						{
-							Record: "enmasse_component_health",
-							Expr:   intstr.FromString("up{job='console',namespace='" + util.GetEnvOrDefault("NAMESPACE", "enmasse-infra") + "'} or on(namespace) (1- absent(up{job='console',namespace='" + util.GetEnvOrDefault("NAMESPACE", "enmasse-infra") + "'}))"),
+		_, err := controllerutil.CreateOrUpdate(ctx, r.client, prometheusRule, func() error {
```
Comment (k-wall) [inline pkg/controller/consoleservice/consoleservice_controller.go:481]: If a user were to enable `ENABLE_MONITORING` then later to disable it again, I think the system should revert to its original state i.e. the PrometheusRule object should be removed.  Should there be an if monitoringEnabled { } else {....} clause, where we attempt gracefully to tidy away what was created?

Comment (robshelly) [reply pkg/controller/consoleservice/consoleservice_controller.go:481]: Creating a separate issue for this as there are a number of monitoring resources installed in different places

**Comment by robshelly** [comment]
@enmasse-ci run tests profile=shared

**Pull request merged**